### PR TITLE
Add documentation on writing a syntax checker

### PR DESCRIPTION
**Pull request by fmdkdd**
Should tick one box off #928.

Feedback and corrections would be welcome :)

### CODE REVIEW

**Comment by fmdkdd** [comment]
Note: I haven't gotten around to use proper RST/Sphinx formatting for the links.  Will do after I RTFM ;)

**Comment by Simplify** [comment]
@fmdkdd this looks great so far!

**Comment by cpitclaudel** [comment]
Ugh, how did I miss that?! Looks beautiful. I think you need to indent the code blocks for github to not get confused, though :)

**Comment by fmdkdd** [comment]
I've done a second pass on this.  I think I've got the RST syntax right.  The syntax highlighting in the Github diff is confused by the backtick syntax of docstring, not much I can do :-)

File: doc/developer/developing.rst [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+.. note::
+
+   Specifically, the above describes the process of *command checkers*, i.e.,
+   checkers that run external programs.  All the checkers defined in
+   ``flycheck-checkers`` are command checkers, but command checkers are actually
+   instances of *generic checkers*.  See :flyc:`flycheck-ocaml` for an example
+   of how to use a generic checker.
+
+.. seealso::
```
Comment (cpitclaudel) [inline doc/developer/developing.rst]: I think the syntax of this seealso is wrong (Github is showing the following two nodes as a code block

**Comment by cpitclaudel** [comment]
Looks great. Approved with one tiny change :)

File: doc/developer/developing.rst (hunk 1 above)
Comment (fmdkdd) [inline doc/developer/developing.rst]: Oh neat!  I didn't know Github could preview RST.  Indeed, it shows a code block, but I've built the documentation locally and the block worked fine.

**Comment by cpitclaudel** [comment]
Thanks! Please merge :)

**Pull request merged**